### PR TITLE
fix wasm exports

### PR DESCRIPTION
**Pull request by MikeHolman**
Previously I had hack which exported last function, just to get up and running. Now you can actually specify your exports properly and Jsrt API returns an object with your exports on it.

[Review on Reviewable](https://reviewable.io/reviews/microsoft/chakracore/337)

### CODE REVIEW

**Comment by dilijev** [comment]
Oh hey! You got some prefast warnings. Good to know the Jenkins prefast check is working :)

File: lib/Runtime/Base/ScriptContext.cpp [hunk 1]
```diff
@@ -1881,9 +1905,9 @@ namespace Js
             {
                 HeapDelete((Wasm::Binary::WasmBinaryReader*)reader);
             }
+
```
Comment (dilijev) [inline lib/Runtime/Base/ScriptContext.cpp]: nit: spaces on blank line

File: test/wasm/basic.js [hunk 2]
```diff
@@ -4,4 +4,4 @@
 //-------------------------------------------------------------------------------------------------------
 
 var a = WScript.LoadWasmFile('basic.wast');
-print(a(11));
\ No newline at end of file
+print(a.a(11));
```
Comment (dilijev) [inline test/wasm/basic.js]: nit: newline at end of file

**Comment by dilijev** [comment]
LGTM (but I'd wait for another set of eyes)

File: lib/Runtime/Base/ScriptContext.cpp [hunk 3]
```diff
@@ -1870,6 +1872,28 @@ namespace Js
...
                 funcObj->SetEnvironment(frameDisplay);
                 localModuleFunctions[i] = funcObj;
+                if (functionArray[i]->wasmInfo->Exported())
+                {
+                    PropertyRecord const * propertyRecord = nullptr;
+                    LPCUTF8 name = wasmScript->module->exports->Lookup(i, nullptr);
+                    if (!name)
+                    {
+                        // TODO: can this happen?
+                        Assert(UNREACHED);
```
Comment (MikeHolman) [inline lib/Runtime/Base/ScriptContext.cpp]: Ah, looks like analysis isn't happy I'm asserting against nullptr but not actually doing anything about it. Fair enough. Guess I'll have to get to that todo sooner than later.

File: lib/Runtime/Base/ScriptContext.cpp [hunk 4]
```diff
@@ -1870,6 +1872,24 @@ namespace Js
...
                 funcObj->SetEnvironment(frameDisplay);
                 localModuleFunctions[i] = funcObj;
+                if (functionArray[i]->wasmInfo->Exported())
+                {
+                    PropertyRecord const * propertyRecord = nullptr;
+                    LPCUTF8 name = functionArray[i]->wasmInfo->GetName();
+                    AnalysisAssertMsg(name, "export function is guaranteed to have name.");
+                    utf8::DecodeOptions decodeOptions = utf8::doDefault;
+                    UINT utf16Len = utf8::ByteIndexIntoCharacterIndex(name, strlen((const char*)name), decodeOptions);
+                    LPCWSTR contents = (LPCWSTR)HeapAlloc(GetProcessHeap(), 0, (utf16Len + 1) * sizeof(WCHAR));
```
Comment (abchatra) [inline lib/Runtime/Base/ScriptContext.cpp]: Who is freeing this?

**Comment by abchatra** [comment]
Looks good to me.

**Pull request merged**